### PR TITLE
allow filtering occ files_external:list to storages applicable for a user

### PR DESCRIPTION
**Pull request by icewind1991**

### CODE REVIEW

**Comment by szaimen** [comment]
@icewind1991 what does this fix or improve?

**Comment by PVince81** [comment]
can be useful in support.
more than once I wanted to ask someone to run this command to see what ext storages are visible only for a problematic user instead of all of them
in many cases you can't infer this from the config when groups are used in "applicable" and the user is member of that

File: apps/files_external/lib/Command/ListCommand.php [hunk 1]
```diff
@@ -92,13 +86,26 @@ protected function configure() {
 				'a',
 				InputOption::VALUE_NONE,
 				'show both system wide mounts and all personal mounts'
+			)->addOption(
+				'for',
+				null,
+				InputOption::VALUE_REQUIRED,
```
Comment (PVince81) [inline apps/files_external/lib/Command/ListCommand.php:92]: should this be optional?

File: apps/files_external/lib/Command/ListCommand.php [hunk 2]
```diff
@@ -92,13 +86,26 @@ protected function configure() {
...
 		/** @var StorageConfig[] $mounts */
-		if ($input->getOption('all')) {
+		if ($for = $input->getOption('for')) {
+			$forUser = $this->userManager->get($for);
+			if (!$forUser) {
+				$output->writeln("<error>User $for not found</error>");
+				return 1;
+			}
+			$mounts = $this->userGlobalService->getAllStoragesForUser($forUser);
+			$userId = self::ALL;
```
Comment (PVince81) [inline apps/files_external/lib/Command/ListCommand.php:107]: a bit quirky but does the job

**Review by PVince81 — APPROVED** [review]
:+1: see minor comments

File: apps/files_external/lib/Command/ListCommand.php [hunk 3]
```diff
@@ -77,13 +86,26 @@ protected function configure(): void {
...
+		} else if ($input->getOption('all')) {
```
Comment (come-nc) [inline apps/files_external/lib/Command/ListCommand.php:108]:
```suggestion
		} elseif ($input->getOption('all')) {
```

File: apps/files_external/lib/Command/ListCommand.php [hunk 4]
```diff
@@ -77,13 +86,26 @@ protected function configure(): void {
 				'a',
 				InputOption::VALUE_NONE,
 				'show both system wide mounts and all personal mounts'
+			)->addOption(
+				'for',
```
Comment (come-nc) [inline apps/files_external/lib/Command/ListCommand.php:90]: It is not clear reading the description what is the difference between `occ files_external:list user` and `occ files_external:list --for user`.
Also `occ files_external:list user1 --for user2` is allowed but will ignore user1?

**Pull request closed without merging**